internal/impl/redpanda/processor_data_transform_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redpanda
+
+//go:generate env GOOS=wasip1 GOARCH=wasm go build -C ../../../resources/testdata/redpanda_data_transforms/uppercase -o $PWD/uppercase.wasm
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func defaultConfig() dataTransformConfig {
+	var cfg dataTransformConfig
+	cfg.maxMemoryPages = 1000
+	cfg.timeout = time.Second
+	return cfg
+}
+
+func TestDataTransformProcessorSerial(t *testing.T) {
+	wasm, err := os.ReadFile("./uppercase.wasm")
+	if os.IsNotExist(err) {
+		t.Skip("skipping as wasm example not compiled, run build.sh to remedy")
```

Comment:
I guess I'd rather fail the tests if this file is missing instead of a silent skip now that we can `go generate` it.

Reply:
I don't feel strongly. Ash do you have opinions? Happy to change them to fail. I'll update the message here either way 😄

Reply:
I've refactored the tests slightly so that we actually build the WASM on the fly. The advantage is that we don't end up with the file sitting stagnant in the repo after the tests run, which can be annoying for some repo activities when it's not automatically cleaned up. It has the disadvantage of slowing the tests down by about a second, but unless this dramatically changes CI test times (I doubt it'll change it) then I'm happy as is.